src/krfmesg/linux/krfmesg.c

```diff
@@ -0,0 +1,82 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <linux/netlink.h>
+#include <unistd.h>
+
+/* Protocol family, consistent in both kernel prog and user prog. */
+#define NETLINK_KRF 28
+/* Multicast group, consistent in both kernel prog and user prog. */
+#define KRF_MGRP 28
+
+int open_netlink(void) {
+  int sock;
+  struct sockaddr_nl addr;
+  int group = KRF_MGRP;
+
+  sock = socket(AF_NETLINK, SOCK_RAW, NETLINK_KRF);
+  if (sock < 0) {
+    printf("Failed to make socket. Is krf module installed?\n");
```

Comment:
You can use `err` and `errx` here and below to avoid this kind of print-and-return pattern (plus it'll be consistent with `krfctl`/`krfexec`)